src/main/java/Domain/ShoppingBasket.java

```diff
@@ -33,6 +36,33 @@ public double getShoppingBasketPrice() {
         return _basketTotalAmount;
     }
 
+    /*
+     * Go through the list of products in the basket and purchase them.
+     * If an exception is thrown, cancel the purchase of all the products that were bought.
+     * This function only updates the item's stock.
+     */
+    public void purchaseBasket() {
+        List<Product> boughtProductList = new ArrayList<>();
```

Comment:
log info: start the purchase progress for id.... user name..... bla bla

Reply:
Done.